### ISSUE TITLE
zip and unzip question

### ISSUE THREAD

**Issue by rastaman111** [ZipArchive/ZipArchive#585]
Good day!  In my application, I use the file system and cannot understand how your product works.  I figured out how to unzip file.txt and how to unzip folder / file.txt.  but I can't figure out how to unify this so that when unzipping the system itself understands where just the file is and gets it, and where is the folder with the files and creates a separate directory

**Comment by rastaman111** [comment]
<img width="903" alt="Снимок экрана 2020-08-12 в 14 46 24" src="https://user-images.githubusercontent.com/15727518/90011595-8a3c9a80-dcaa-11ea-8b7a-525da22bc4a5.png">

first func unzipping folder
second func unzipping file

Рow to make the system understand before unzipping the file what is in it?

**Comment by willson-chen** [comment]
@rastaman111 Hope the example in the link below can help you.
https://github.com/ZipArchive/ZipArchive/blob/546b99c5c0f7e0c5761960c8ffd486f2ef7610d7/Example/ObjectiveCExample/ViewController.m#L66

https://github.com/ZipArchive/ZipArchive/blob/546b99c5c0f7e0c5761960c8ffd486f2ef7610d7/Example/ObjectiveCExample/ViewController.m#L157

**Comment by rastaman111** [comment]
Yes, I saw you as an example, it shows how to unzip a folder with files, but it does not show how to unzip, if in the ZIP which contains only there, for example, one file and therefore when it is unzipped it creates a directory under the name of this file which is in it and for this there is a problem

**Comment by willson-chen** [comment]
Do you mean that you want to get the zip package information before unizp?

**Comment by willson-chen** [comment]
Maybe you can pass in the zip name to `fileName`, such as `abc.zip`, pass `abc` to `fileName`.

**Comment by rastaman111** [comment]
let resourceKeys: [URLResourceKey] = [.creationDateKey, .isDirectoryKey]
        let enumerator = FileManager.default.enumerator(at: file.originalUrl,
                                                        includingPropertiesForKeys: resourceKeys,
                                                        options: [.skipsHiddenFiles], errorHandler: { (url, error) -> Bool in
                                                            print("directoryEnumerator error at \(url): ", error)
                                                            return true
        })!

        let documentsURL = DataManager.getDocumentsFolderURL()

        for case let fileURL as URL in enumerator {
            guard fileURL.pathExtension == "mp3" || fileURL.pathExtension == "m4a" || fileURL.pathExtension == "m4b" else { continue }

            let destinationURL = documentsURL.appendingPathComponent(fileURL.lastPathComponent)
            try? FileManager.default.moveItem(at: fileURL, to: destinationURL)
        }

        // Delete directory
        try? FileManager.default.removeItem(at: file.originalUrl)


i want to try this code, it iterates over all items in zip

**Comment by rastaman111** [comment]
@willson-chen The top function works well both for a simple file and for a folder with files
